File: src/main/java/com/wxl/dyttcrawler/properties/CrawlerProperties.java
package com.wxl.dyttcrawler.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.time.Duration;
import java.util.Collections;
import java.util.List;
import java.util.Map;

/**
 * Create by wuxingle on 2020/5/16
 * 爬虫配置
 */
@Data
@ConfigurationProperties(prefix = "crawler")
public class CrawlerProperties {

    private String taskId;

    private String firstUrl = "https://www.dytt8.net";

    private String charset = "gbk";

    private int maxThreads = 5;

    private SiteProperties site = new SiteProperties();

    @Data
    public static class SiteProperties {

        private String userAgent;

        private Duration sleepTime = Duration.ofSeconds(1);

        private Integer retryTimes = 1;

        private Duration retrySleepTime = Duration.ofSeconds(1);

        private Duration timeout = Duration.ofSeconds(10);

        private List<Integer> acceptStatusCode = Collections.singletonList(200);

        private Map<String, String> headers = Collections.emptyMap();

        private boolean disableCookie = true;
    }
}

File: src/main/kotlin/com/wxl/dyttcrawler/properties/CrawlerProperties.kt
package com.wxl.dyttcrawler.properties

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * Create by wuxingle on 2021/10/02
 * 爬虫配置
 */
@ConfigurationProperties(prefix = "crawler")
data class CrawlerProperties(

    /**
     * 任务id
     */
    var taskId: String? = null,

    /**
     * 拉取起始地址
     */
    var startUrl: String = "https://www.dytt8.net",

    /**
     * 匹配的域名
     */
    var allowDomains: List<String> = emptyList(),

    /**
     * 爬虫任务底层线程数
     */
    var maxThreads: Int = 5,

    /**
     * 爬虫任务并发度
     */
    var concurrentNum: Int = 1
)

